main/src/ca/uwaterloo/flix/tools/Tester.scala

```diff
@@ -34,28 +34,36 @@ object Tester {
   /**
     * Runs all tests.
     */
-  def run(filters: List[Regex], compilationResult: CompilationResult)(implicit flix: Flix): Unit = {
+  def run(filters: List[Regex], compilationResult: CompilationResult)(implicit flix: Flix): Result[Unit, Int] = {
     //
     // Find all test cases (both active and ignored).
     //
     val tests = getTestCases(filters, compilationResult)
 
     // Start the TestRunner and TestReporter.
     val queue = new ConcurrentLinkedQueue[TestEvent]()
-    val reporter = new TestReporter(queue, tests)
+    val failureQueue = new ConcurrentLinkedQueue[Int]()
```

Comment:
I would prefer if we did something else. Can we not have TestReporter return a future with the result?

Reply:
Actually I don't see a reason why `TestReporter` runs in his own thread, if the main thread is blocking in the `join`s anyway. I propose just using the `TestReporter.run` function as a "normal" one.
See https://github.com/flix/flix/pull/5855/commits/d225d9fa9f9f409a11c470553ebfc4d8b16a9899